integrations/oci/oci-secrets-config-source/src/main/java/io/helidon/integrations/oci/secrets/configsource/OciSecretsConfigSourceProvider.java

```diff
@@ -0,0 +1,526 @@
+                    }
+                    return Optional.of(NodeContent.builder()
+                                       .node(onb.build())
+                                       .build());
+                };
+            }
+        }
+
+
+        /*
+         * Instance methods.
+         */
+
+
+        @Deprecated // For use by the Helidon Config subsystem only.
+        @Override // PollableSource
+        public boolean isModified(Instant instant) {
+            Instant i = this.mostDistantExpirationInstant;
+            return i == null || i.isBefore(instant == null ? now() : instant);
+        }
+
+        @Deprecated // For use by the Helidon Config subsystem only.
+        @Override // NodeConfigSource
+        public Optional<NodeContent> load() {
+            return this.loader.get();
+        }
+
+        @Deprecated // For use by the Helidon Config subsystem only.
+        @Override // PollableSource
+        public Optional<PollingStrategy> pollingStrategy() {
+            return super.pollingStrategy();
+        }
+
+        private Optional<NodeContent> absentNodeContent() {
+            this.mostDistantExpirationInstant = null; // volatile write
+            return ABSENT_NODE_CONTENT;
+        }
+
+
+        /*
+         * Static methods.
+         */
+
+
+        private static Builder builder() {
+            return new Builder();
+        }
+
+        private static void completeAllSecretsTasks(Collection<? extends Callable<Void>> tasks, AutoCloseable secrets) {
+            RuntimeException re = null;
+            try (ExecutorService es = newVirtualThreadPerTaskExecutor()) {
+                for (Future<?> future : es.invokeAll(tasks)) {
+                    try {
+                        future.get();
+                    } catch (RuntimeException e) {
+                        if (re == null) {
+                            re = e;
+                        } else {
+                            re.addSuppressed(e);
+                        }
+                    } catch (Exception e) {
+                        if (e instanceof InterruptedException) {
+                            Thread.currentThread().interrupt();
+                        }
+                        if (re == null) {
+                            re = new IllegalStateException(e.getMessage(), e);
+                        } else {
+                            re.addSuppressed(e);
+                        }
+                    }
+                }
+            } catch (RuntimeException e) {
+                re = e;
+            } catch (Exception e) {
+                if (e instanceof InterruptedException) {
+                    Thread.currentThread().interrupt();
+                }
+                re = new IllegalStateException(e.getMessage(), e);
+            } finally {
+                try {
+                    secrets.close();
+                } catch (RuntimeException e) {
+                    if (re == null) {
+                        throw e;
```

Comment:
you should never throw intentionally from a finally block. 
This whole construct is a bit strange, but it could be changed by assigning to `re` or suppressing, and the final `if (re != null) thorw re` could be done after the `finally` block is finished.

Reply:
Done